### ISSUE TITLE
Extremely simple http-01 in apache

### ISSUE THREAD

**Issue by kuba** [certbot/certbot#1272]
How about implementing `http-01` for Apache? After writing #1268 I realized that the authenticator would not even have to touch users config - just drop a file in `/etc/apache2/conf-enabled/` with the following contents:

```
<IfModule mod_headers.c>
    <LocationMatch "/.well-known/acme-challenge/*">
        Header set Content-Type "application/jose+json"
        DocumentRoot /var/lib/letsencrypt/apache-http01-root
    </LocationMatch>
</IfModule>
```

reload the daemon, drop necessary resources in `/var/lib/letsencrypt/apache-http01-root`, and voila!

The biggest advantage is that we might even get away without read access (parsing) of the existing configs, not even mentioning write access :) That means less scope for errors (a lot of users don't even get to the authorization stage, because of MisconfigurationErrors raised in the parsing step).

Possible problems:
- users don't have `/etc/apache/conf-enabled` by default
- `DocumentRoot` seems to be allowed there (https://httpd.apache.org/docs/2.2/mod/core.html#documentroot), but if it's not, then we can drop the stanza and try to read the config file and find existing `public_html`

@jdkasten?

**Comment by chriscroome** [comment]
Altough Apache doesn't appear to complain about having a `DocumentRoot` within a `LocationMatch` I'm not sure it is supposed to work -- there is no `directory` context for `DocumentRoot`:

> DocumentRoot Directive
> Context:	server config, virtual host
> https://httpd.apache.org/docs/2.4/mod/core.html#documentroot

I'd suggest using an `Alias` for this:

```apache
Alias "/.well-known/acme-challenge" "/var/lib/letsencrypt/apache-http01-root/.well-known/acme-challenge"
<IfModule mod_headers.c>
    <LocationMatch "/.well-known/acme-challenge/*">
        Header set Content-Type "application/jose+json"
    </LocationMatch>
</IfModule>

**Comment by not-implemented** [comment]
Nice solution. I just miss a "Require all granted" here. (And the mod_headers check can be moved deeper):

```
Alias "/.well-known/acme-challenge" "/var/lib/letsencrypt/apache-http01-root/.well-known/acme-challenge"

<Directory "/var/lib/letsencrypt/apache-http01-root/.well-known/acme-challenge">
    Require all granted

    <IfModule mod_headers.c>
        Header set Content-Type "application/jose+json"
    </IfModule>
</Directory>
```

**Comment by bmw** [comment]
We implemented a not so simple version of this in 0.21.0 due to Let's Encrypt removing TLS-SNI-01 support.
